### PR TITLE
[DOC] Tidy up referencing in doc strings

### LINKED ISSUE #685: [BUG] Weird referencing numbering in the API
### Describe the bug

in some modules, e.g. deep learning and clustering, referenced paper numbers do not appear correctly. It is probably some weird encoding between unix/mac/windows conversions. 

### Steps/Code to reproduce the bug

https://aeon-toolkit--673.org.readthedocs.build/en/673/api_reference/classification.html

### Expected results

should be numbers e.g [1]

### Actual results

image

### Versions

_No response_

### PR DESCRIPTION
**Pull request by TonyBagnall**
Fixes #685 

sorts of references so not in weird format and link

### CODE REVIEW

File: aeon/classification/deep_learning/encoder.py [hunk 1]
```diff
@@ -20,7 +20,7 @@ class EncoderClassifier(BaseDeepClassifier):
     """
     Establish the network structure for an Encoder.
 
-    Adapted from the implementation used in [1]
+    Adapted from the implementation used in [1]_
```
Comment (dguijo) [inline aeon/classification/deep_learning/encoder.py:23]: Final dot?

File: aeon/classification/dictionary_based/_cboss.py [hunk 2]
```diff
@@ -25,8 +25,8 @@ class ContractableBOSS(BaseClassifier):
     """
     Contractable Bag of Symbolic Fourier Approximation Symbols (cBOSS).
 
-    Implementation of BOSS Ensemble from Schäfer (2015) with refinements
-    described in Middlehurst, Vickers and Bagnall (2019). [1, 2]_
+    Implementation of BOSS Ensemble from [1]_ with refinements
+    described in Middlehurst, Vickers and Bagnall (2019). [2]_
```
Comment (dguijo) [inline aeon/classification/dictionary_based/_cboss.py:29]: "... from [1]_ with refinements described in [2]_." ?

File: aeon/classification/dictionary_based/_tde.py [hunk 3]
```diff
@@ -611,7 +611,7 @@ class IndividualTDE(BaseClassifier):
     Single TDE classifier, an extension of the Bag of SFA Symbols (BOSS) model.
 
     Base classifier for the TDE classifier. Implementation of single TDE base model
-    from Middlehurst (2021). [1]_
+    from [1]_
```
Comment (dguijo) [inline aeon/classification/dictionary_based/_tde.py:615]: final dot?

File: aeon/classification/hybrid/_hivecote_v1.py [hunk 4]
```diff
@@ -31,7 +31,7 @@ class HIVECOTEV1(BaseClassifier):
 
     An ensemble of the STC, TSF, RISE and cBOSS classifiers from different feature
     representations using the CAWPE structure as described in [1]_. The default
-    implementation differs from the one described in [1]_, in that the STC component
+    implementation differs from the one described in [1], in that the STC component
```
Comment (dguijo) [inline aeon/classification/hybrid/_hivecote_v1.py:34]: [1]_ ?

File: aeon/classification/interval_based/_tsf.py [hunk 5]
```diff
@@ -18,7 +18,8 @@
 class TimeSeriesForestClassifier(BaseIntervalForest, BaseClassifier):
     """Time series forest (TSF) classifier.
 
-    A time series forest is an ensemble of decision trees built on random intervals.
+    A time series forest is an ensemble of decision trees built on random intervals [
+    1]_.
```
Comment (dguijo) [inline aeon/classification/interval_based/_tsf.py:22]: [1]_ in the same line?

**Review by dguijo — CHANGES_REQUESTED** [review]
minor typos. What a tedious work, Tony! Thanks!

**Comment by TonyBagnall** [comment]
> minor typos. What a tedious work, Tony! Thanks!

lol, I have a strangely large capacity for tedious work like this, its the secret of my success :)

**Comment by TonyBagnall** [comment]
thanks david, typos corrected.

**Review by dguijo — APPROVED** [review]
lgtm. Thank you, Tony!

**Pull request merged**